src/IssueViz/Editor/QuickActions/IssueLocationActionsSource.cs

```diff
@@ -0,0 +1,123 @@
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Tagging;
+using SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging;
+using SonarLint.VisualStudio.IssueVisualization.Editor.SelectedIssueTagging;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Selection;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Editor.QuickActions
+{
+    internal sealed class IssueLocationActionsSource : ISuggestedActionsSource
+    {
+        private readonly IAnalysisIssueSelectionService selectionService;
+        private readonly ITagAggregator<IIssueLocationTag> issueLocationsTagAggregator;
+        private readonly ITagAggregator<ISelectedIssueLocationTag> selectedIssueLocationsTagAggregator;
+
+        public IssueLocationActionsSource(IBufferTagAggregatorFactoryService bufferTagAggregatorFactoryService, ITextBuffer textBuffer, IAnalysisIssueSelectionService selectionService)
+        {
+            this.selectionService = selectionService;
+
+            issueLocationsTagAggregator = bufferTagAggregatorFactoryService.CreateTagAggregator<IIssueLocationTag>(textBuffer);
+            issueLocationsTagAggregator.TagsChanged += TagAggregator_TagsChanged;
+
+            selectedIssueLocationsTagAggregator = bufferTagAggregatorFactoryService.CreateTagAggregator<ISelectedIssueLocationTag>(textBuffer);
+            selectedIssueLocationsTagAggregator.TagsChanged += TagAggregator_TagsChanged;
+        }
+
+        public event EventHandler<EventArgs> SuggestedActionsChanged;
+
+        public IEnumerable<SuggestedActionSet> GetSuggestedActions(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
+        {
+            var allActions = new List<ISuggestedAction>();
+
+            if (IsOnIssueVisualization(range, out var issueVisualizations))
+            {
+                var actions = issueVisualizations.Select(x => new SelectIssueVisualizationAction(selectionService, x));
```

Comment:
Will we show the light bulb for an issue that is already selected?

Reply:
Yes, I was debating about this. If there are multiple issues on the same line, one of which is selected and another isn't, I thought it might be confusing to the user -- as he might think that the "Show" action is for the issue that he already sees, and not a different one. This way it's more clear that there are 2+ issues on the same row.

Reply:
So currently we'll always suggest a pointless action in what is probably the most common case of a single issue already being selected? That could be annoying.

Two issues on the same line for the same rule seems pretty uncommon.

Reply:
Yep possibly. Ok, I'll add a check that it's not already selected.